core/miner.go

```diff
@@ -36,27 +36,29 @@ import (
 
 // Miner creates blocks and searches for proof-of-work values.
 type Miner struct {
-	worker   *worker
-	coinbase common.Address
-	hc       *HeaderChain
-	engine   consensus.Engine
-	startCh  chan common.Address
-	stopCh   chan struct{}
-	logger   *log.Logger
+	worker            *worker
+	primaryCoinbase   common.Address
+	secondaryCoinbase common.Address
+	hc                *HeaderChain
+	engine            consensus.Engine
+	startCh           chan []common.Address
+	stopCh            chan struct{}
+	logger            *log.Logger
 }
 
 func New(hc *HeaderChain, txPool *TxPool, config *Config, db ethdb.Database, chainConfig *params.ChainConfig, engine consensus.Engine, isLocalBlock func(block *types.WorkObject) bool, processingState bool, logger *log.Logger) *Miner {
 	miner := &Miner{
-		hc:       hc,
-		engine:   engine,
-		startCh:  make(chan common.Address),
-		stopCh:   make(chan struct{}),
-		worker:   newWorker(config, chainConfig, db, engine, hc, txPool, isLocalBlock, true, processingState, logger),
-		coinbase: config.Etherbase,
+		hc:                hc,
+		engine:            engine,
+		startCh:           make(chan []common.Address, 2),
+		stopCh:            make(chan struct{}),
+		worker:            newWorker(config, chainConfig, db, engine, hc, txPool, isLocalBlock, true, processingState, logger),
+		primaryCoinbase:   config.PrimaryCoinbase,
+		secondaryCoinbase: config.SecondaryCoinbase,
 	}
 	go miner.update()
 
-	miner.Start(miner.coinbase)
+	miner.Start(miner.primaryCoinbase, miner.secondaryCoinbase)
```

Comment:
Can we just delete this

Reply:
we need this because this is how we set teh coinbase in the worker env

Reply:
No the coinbase is in the config.Etherbase

Reply:
we have two coinbases now, and i know how we are doing this is not ideal, we can change it in the future

Reply:
deleting this

Reply:
Thx